core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java

```diff
@@ -420,10 +419,13 @@ public InteractiveTransactionResponse deposit(Sep10Jwt token, Map<String, String
     }
 
     String quoteId = depositRequest.get("quote_id");
+    AssetInfo sellAsset = assetService.getAssetByName(depositRequest.get("source_asset"));
     if (quoteId != null) {
-      AssetInfo sellAsset = assetService.getAssetByName(depositRequest.get("source_asset"));
-      this.validatedAndPopulateQuote(
+      validateAndPopulateQuote(
           quoteId, sellAsset, asset, strAmount, builder, DEPOSIT.toString(), txnId);
+    } else {
+      builder.amountExpected(strAmount);
+      if (sellAsset != null) builder.amountInAsset(sellAsset.getSep38AssetName());
```

Comment:
Should we also call `builder.amountIn("0")` for consistency with SEP-6 impl?

Reply:
oh yeah. Updated.